### PR TITLE
feat: Enable Rule CA1304

### PR DESCRIPTION
**Pull request by workgroupengineering**
## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues

Part of #8944

### CODE REVIEW

File: src/Avalonia.Controls/MenuItemAccessKeyHandler.cs [hunk 1]
```diff
@@ -88,9 +88,9 @@ protected virtual void OnTextInput(object? sender, TextInputEventArgs e)
         {
             if (!string.IsNullOrWhiteSpace(e.Text))
             {
-                var text = e.Text.ToUpper();
+                var text = e.Text.ToUpperInvariant();
                 var focus = _registered
-                    .FirstOrDefault(x => x.Item1 == text && x.Item2.IsEffectivelyVisible)?.Item2;
+                    .FirstOrDefault(x => x.Element.IsEffectivelyVisible && x.Item1 == text).Element;
```
Comment (maxkatz6) [inline src/Avalonia.Controls/MenuItemAccessKeyHandler.cs:93]: Why was order changed here?
String comparison is way faster than reading this property: https://github.com/AvaloniaUI/Avalonia/blob/18b1fb74fe7e0393b7768286d01b6e16b50acce9/src/Avalonia.Base/Visual.cs#L192

Comment (maxkatz6) [reply src/Avalonia.Controls/MenuItemAccessKeyHandler.cs:93]: Also, consider string.Equals with Ordinal or Invariant comparison, as by default current culture comparison is used.

Comment (workgroupengineering) [reply src/Avalonia.Controls/MenuItemAccessKeyHandler.cs:93]: I inverted the order it is useless to check equality if the MenuIItem is disabled.

Comment (maxkatz6) [reply src/Avalonia.Controls/MenuItemAccessKeyHandler.cs:93]: Yes, but it's way faster to check equality, that check if item is disabled. Check the link, this property also traverses the tree.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026899-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-nightly/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026963-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-nightly/nuget/v3/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026971-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-nightly/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Base/Input/KeyGesture.cs [hunk 2]
```diff
@@ -143,7 +144,7 @@ static void Plus(StringBuilder s)
         // TODO: Move that to external key parser
         private static Key ParseKey(string key)
         {
-            if (s_keySynonyms.TryGetValue(key.ToLower(), out Key rv))
+            if (s_keySynonyms.TryGetValue(key.ToLower(CultureInfo.InvariantCulture), out Key rv))
```
Comment (robloo) [inline src/Avalonia.Base/Input/KeyGesture.cs:147]: Note for the future there is a `.ToLowerInvariant()` that should be used in places like this.

**Comment by robloo** [comment]
I like the invariant string comparison fixes. These types of rules are always good to enable :)

**Pull request merged**